src/Param.cc

```diff
@@ -65,54 +65,66 @@ Param::Param(const std::string &_key, const std::string &_typeName,
              const std::string &_description)
   : dataPtr(new ParamPrivate)
 {
-  this->dataPtr->key = _key;
-  this->dataPtr->required = _required;
-  this->dataPtr->typeName = _typeName;
-  this->dataPtr->description = _description;
-  this->dataPtr->set = false;
-  this->dataPtr->ignoreParentAttributes = false;
-  this->dataPtr->defaultStrValue = _default;
+  sdf::Errors errors;
+  this->dataPtr->Init(_key, _typeName, _default, _required,
+                      errors, _description);
 
-  SDF_ASSERT(
-      this->dataPtr->ValueFromStringImpl(
-          this->dataPtr->typeName,
-          _default,
-          this->dataPtr->defaultValue),
-      "Invalid parameter");
-  this->dataPtr->value = this->dataPtr->defaultValue;
-  this->dataPtr->strValue = std::nullopt;
+  if(!errors.empty())
+  {
+    for (unsigned int i = 0; i < errors.size() - 1; ++i)
+    {
+      sdferr << errors[i].Message() + "\n";
+    }
+
+    SDF_ASSERT(false, errors[errors.size() - 1].Message());
```

Comment:
nit: You can use `errors.back()` instead.

Reply:
Also changed here: https://github.com/ignitionrobotics/sdformat/blob/a7ee2929e5e51bb578ef9a0f85ad18df76d9fc1d/src/Param.cc#L112.